.github/workflows/new-bugs-announce-notifier.yml

```
@@ -19,13 +19,13 @@ jobs:
       - name: Send notification
         uses: actions/github-script@v6
         env:
-          MAILGUN_API_KEY: ${{ secrets.PSF_MAILGUN_KEY }}
+          MAILGUN_API_KEY: ${{ secrets.MAILGUN_PYTHON_ORG_MAILGUN_KEY }}
```

I know this reads very redundantly, but its the mailgun key for mailgun.python.org 😅

---

Probably there is no need to clarify the exact subdomain, just a site affinity:

```suggestion
          MAILGUN_API_KEY: ${{ secrets.PYTHON_ORG_MAILGUN_KEY }}
```

---

there are other subdomains, so it does make it much easier to reason about when it comes time to rotate or manage the secret in the future... I'd rather be explicit. (this matches configs for the other tenants in the mailgun account).